swap_meet/vendor.py

```diff
@@ -1,2 +1,125 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+    
+    def add(self, item):
+        """Adds an item to the instance's inventory and returns it."""
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        """
+        Removes items from the instance's inventory.
+
+        Parameter: 
+        item (str): Inventory item to be removed.
+
+        Returns:
+        bool: True if item is inventory, False otherwise.
+        """
+
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_category(self, specific_category):
+        """
+        Adds the items in the instance's inventory that are of a specific category.
+
+        Parameters: 
+        specific_category (str): A category.
+
+        Returns:
+        category_matches (list): Items in the inventory matching the category.
+        """
+
+        category_matches = []
+        for item in self.inventory:
+            if item.category == specific_category:
+                category_matches.append(item)
+        return category_matches
+
+    def swap_items(self, friend, my_item, their_item):
+        """
+        Swaps items in vendor's inventory and friend's inventory.
+
+        Parameters: 
+        friend (class): Vendor instance of friend.
+        my_item (str): Category the vendor wants to receive.
+        their_item (str): Category friends wants to receive.
+
+        Returns:
+        (bool): True if swapping occurs, False if not.
+        """
+
+        if my_item in self.inventory and their_item in friend.inventory:
+            friend.inventory.append(my_item)
+            self.inventory.remove(my_item)
+            
+            self.inventory.append(their_item)
+            friend.inventory.remove(their_item)
+            return True
+        return False
+    
+    def swap_first_item(self, friend):
+        '''
+        Removes first item from a vendor's inventory and a friend's inventory.
+
+        Parameter:
+        friend (class): An instance of another Vendor.
+
+        Returns: 
+        (bool): True is first items the Vendor and the friend swap items, False otherwise.
+        '''
+        if self.inventory and friend.inventory:
+            friend.inventory.append(self.inventory[0])
+            self.inventory.remove(self.inventory[0])
+
+            self.inventory.append(friend.inventory[0])
+            friend.inventory.remove(friend.inventory[0])
+            return True
```

Comment:
We could reuse the code you wrote to swap items here! 😄 Both the solution above and reusing the `swap_items` function winds up being O(n), mostly due to removing the items from the inventory. 
```suggestion
            my_item = self.inventory[0]
            friend_item = friend.inventory[0]
            return self.swap_items(friend, my_item, friend_item)
```

We could also swap the items in place to have O(1) runtime with something like:
```python
self.inventory[0], friend_vendor.inventory[0] = friend_vendor.inventory[0], self.inventory[0]
```